### PR TITLE
use -mavx clang style options to enable SIMD for clang-cl

### PR DESCRIPTION
**Pull request by wangyoucao577**
- Only use `/arch:AVX` MSVC style options to enable SIMD when strictly MSVC cl;        
- Use `-mavx` gcc/clang style options to enable SIMD when possible, for example clang-cl installed in MSVC;      
- Enable `C11` for `clang-cl`;        
- Enable cpuinfo for `clang-cl`;

Closes https://github.com/xiph/opus/issues/256

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -389,13 +389,13 @@ if(NOT OPUS_DISABLE_INTRINSICS)
     if(OPUS_X86_MAY_HAVE_SSE)
       add_sources_group(opus celt ${celt_sources_sse})
       target_compile_definitions(opus PRIVATE OPUS_X86_MAY_HAVE_SSE)
-      if(NOT MSVC)
+      if(NOT (MSVC AND CMAKE_C_COMPILER_ID STREQUAL "MSVC"))
```
Comment (xnorpx) [inline CMakeLists.txt:392]: Maybe easier to make a variable for the case when using MSVC compiler frontend and not.

**Comment by xnorpx** [comment]
@wangyoucao577 can you add the same check for all other compiler flags regarding clang-cl frontend. (fast-math etc)

File: CMakeLists.txt (hunk 1 above)
Comment (wangyoucao577) [reply CMakeLists.txt]: Good point! Updated.

**Comment by wangyoucao577** [comment]
> @wangyoucao577 can you add the same check for all other compiler flags regarding clang-cl frontend. (fast-math etc)

I checked all `MSVC` related options in current cmake, unfortunately `-ffast-math` can't work on `clang-cl`, it must be `/fp:fast`. Supported options(c11, cpuinfo) has been updated.

**Comment by jmvalin** [comment]
What's the status of this PR?

**Comment by xnorpx** [comment]
Master branch is deleted and all pr's against master is closed. If this change is still relevant please reopen and repoint your PR to main branch.

**Pull request closed without merging**